full-service/src/json_rpc/wallet.rs

```diff
@@ -590,6 +590,32 @@ where
                 txo: Txo::from(&result),
             }
         }
+        JsonCommandRequest::split_txo {
+            txo_id,
+            output_values,
+            subaddress_index,
```

Comment:
Part of me thinks we should maybe pass a public address rather than a subaddress_index, so that you could potentially `split_txo` to an abitrary address, but that's not actually required by the spec. Alternatively, would maybe rename this `destination_subaddress_index` to make it clear that it's not the source of the funds.

Reply:
I was thinking that as well, but my gut is saying that this should restrict splitting a txo to remain on the account that owns it. I guess one could still perform a check to make sure the subaddress belongs to the account that owns the txo, but this seemed more straightforward. Do you have a strong preference either way, or maybe even allowing either an index or a subaddress_b58?

Reply:
To follow that up, if a user wants to _move_ the txo to a different subaddress that _may or may not_ be a part of the account, then the normal build_transaction or build_and_submit_transaction endpoints would be appropriate

Reply:
That works for me :) Let's restrict `split_txo` to remain in the same account, and just provide the index